lib/buildkite/test_collector/library_hooks/rspec.rb

```diff
@@ -12,7 +12,8 @@
   config.before(:suite) do
     config.add_formatter Buildkite::TestCollector::RSpecPlugin::Reporter
 
-    Buildkite::TestCollector.safe { Buildkite::TestCollector::Uploader.configure }
+    Thread.current[:_buildkite_send_queue] = []
+    Thread.current[:_buildkite_upload_threads] = []
```

Comment:
I'm just putting these variable in the thread local variable space because I haven't created a new session class to store stuff like this. But all the code that runs in these various config blocks are in the main thread, and so it is shared